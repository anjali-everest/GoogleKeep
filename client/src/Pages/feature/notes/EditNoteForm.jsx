import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux"
import { updateIsEditingNote, selectEditingNoteId, selectNoteById } from "./notesSlice"
import { fetchNotes, updateOneNote } from "./ThunkHandler"

export const EditNoteForm = () => {
    const noteId = useSelector(selectEditingNoteId)
    const note = useSelector(state => selectNoteById(state, parseInt(noteId)))

    const dispatch = useDispatch()

    const [title, setTitle] = useState(note.title)
    const [content, setContent] = useState(note.content)

    const onTitleChanged = e => setTitle(e.target.value)
    const onContentChanged = e => setContent(e.target.value)
    const onSaveNoteClicked = () => {
        if (title && content) {
            dispatch(updateOneNote({ id: noteId, title, content }))
            dispatch(updateIsEditingNote())
            dispatch(fetchNotes())
        }
    }

    return (
        <div className="editNoteModel" >
            <div>
                <textarea
                    value={title}
                    onChange={onTitleChanged}
                />
                <textarea
                    value={content}
                    onChange={onContentChanged}
                />
                <button onClick={onSaveNoteClicked}>Save</button>
            </div>
        </div >
    );
};
