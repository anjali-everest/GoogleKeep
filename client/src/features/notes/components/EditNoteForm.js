import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  updateIsEditingNote,
  selectEditingNoteId,
  selectNoteById,
} from "../notesSlice";
import { fetchNotes, updateOneNote } from "../thunkHandler";

export const EditNoteForm = () => {
  const noteId = useSelector(selectEditingNoteId);
  const note = useSelector((state) => selectNoteById(state, parseInt(noteId)));

  const dispatch = useDispatch();

  const [title, setTitle] = useState(note.title);
  const [content, setContent] = useState(note.content);

  const onTitleChanged = (e) => setTitle(e.target.value);
  const onContentChanged = (e) => setContent(e.target.value);
  const onSaveNoteClicked = () => {
    if (isUpdateNeeded()) {
      dispatch(updateOneNote({ id: noteId, title, content }));
      dispatch(fetchNotes());
    }
    dispatch(updateIsEditingNote());
  };
  const isUpdateNeeded = () => {
    return note.title !== title || note.content !== content;
  };

  return (
    <div className="edit-note-form">
      <div className="model">
        <textarea
          className="note-title"
          value={title}
          onChange={onTitleChanged}
        />
        <textarea
          className="note-content"
          value={content}
          onChange={onContentChanged}
        />
        <button className="note-save-button" onClick={onSaveNoteClicked}>
          Save
        </button>
      </div>
    </div>
  );
};
