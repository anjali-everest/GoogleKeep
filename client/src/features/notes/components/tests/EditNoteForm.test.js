import React from "react";
import { mount } from "enzyme";
import { EditNoteForm } from "../EditNoteForm";
import configureMockStore from "redux-mock-store";
import * as ReactReduxHooks from "react-redux";
const mockStore = configureMockStore();

describe("EditNoteForm tests", () => {
  const note = { id: 1, title: "title", content: "content" };
  const store = mockStore({
    notes: [note],
    status: "idle",
    error: null,
    editingNoteId: 1,
    isEditingNote: true,
    deletingNoteId: null,
  });

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(() => store.getState());
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  it("Should render without crashing", () => {
    mount(<EditNoteForm />);
  });

  it("should match the EditNoteForm snapshot", function () {
    const editNoteForm = mount(<EditNoteForm />);
    expect(editNoteForm).toMatchSnapshot();
  });
});
