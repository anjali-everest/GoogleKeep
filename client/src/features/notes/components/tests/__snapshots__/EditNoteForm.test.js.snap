// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNoteForm tests should match the EditNoteForm snapshot 1`] = `ReactWrapper {}`;
